app/services/account_reset/track_irs_event.rb

```diff
@@ -0,0 +1,36 @@
+# Mixin for account reset event tracking
+# Assumes these methods exist on the including class:
+# - sp
+# - success?
+# - errors
+# - request
+# - analytics
+module AccountReset::TrackIrsEvent
+  def track_irs_event
+    irs_attempts_api_tracker.account_reset_account_deleted(
+      success: success,
+      failure_reason: event_failure_reason.presence,
+    )
+  end
+
+  def irs_attempts_api_tracker
+    @irs_attempts_api_tracker ||= IrsAttemptsApi::Tracker.new(
+      session_id: nil,
+      request: request,
+      user: user,
+      sp: sp,
+      cookie_device_uuid: cookies[:device],
+      sp_request_uri: nil,
+      enabled_for_session: sp.irs_attempts_api_enabled?,
```

Comment:
@n1zyy this just reminder me of this line in your PR:

https://github.com/18F/identity-idp/pull/7951/files#diff-0414a45947b697e51d740c5ba32eb6c6445f4c795807250ce68106de5ff7e414R133

We have the `enabled_for_session` so the tracker will no-op if it's not enabled, we can do that instead of having a. nillable tracker